File: src/somelib_client.py
import requests

host = 'http://0.0.0.0:8080'


def g(action, *args):
    url = host+'/'+action
    for arg in args:
        url += '/'+str(arg)
    r = requests.get(url)
    return r.json()


def mouse(cheese, trap, cat):
    return g('mouse', cheese, trap, cat)['return_value']


def dog(bone):
    return g('dog', bone)['return_value']


if __name__ == '__main__':
    pass

File: src/somelib_client.js
var request = require('request');

var clientApi = {
    host : 'http://0.0.0.0:8080',
    g : function(action, args){
        if(arguments.length < 2) {
            return {};
        } else {
            var options = {
                'url': this.host,
                'method': 'GET'
            };
            var callback = arguments[arguments.length - 1];
            for(var i = 0; i < arguments.length - 1; i++) {
                options['url'] += '/' + arguments[i];
            }
            request(options, function(err, res, body) {
                callback(err, res, body);
            });
        }
    },
    mouse: function(cheese, trap, cat, cb){
        this.g('mouse', cheese, trap, cat, cb);
    },
    dog: function(bone, cb){
        this.g('dog', bone, cb);
    }
};
